debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/launchConfigurations/LaunchConfigurationTabGroupViewer.java

```diff
@@ -1449,19 +1452,39 @@ protected void handleTabSelected() {
 		if (fDisposingTabs || fInitializingTabs) {
 			return;
 		}
+		int oldCurrentTabIndex = fCurrentTabIndex;
+		fCurrentTabIndex = fTabFolder.getSelectionIndex();
+
 		ILaunchConfigurationTab[] tabs = getTabs();
-		if (fCurrentTabIndex == fTabFolder.getSelectionIndex() || tabs == null || tabs.length == 0 || fCurrentTabIndex > (tabs.length - 1)) {
+		if (oldCurrentTabIndex == fCurrentTabIndex || tabs == null || tabs.length == 0
+				|| oldCurrentTabIndex > (tabs.length - 1)) {
 			return;
 		}
-		if (fCurrentTabIndex != -1) {
-			ILaunchConfigurationTab tab = tabs[fCurrentTabIndex];
-			ILaunchConfigurationWorkingCopy wc = getWorkingCopy();
-			if (wc != null) {
-				tab.deactivated(wc);
-				getActiveTab().activated(wc);
-			}
+
+		deactivateTab(oldCurrentTabIndex);
+
+		activateCurrentlyActiveTab();
+	}
+
+	private void deactivateTab(int tabIndex) {
```

Comment:
Misleading method name: This does not deactivate a tab, but only notifies a tab that it was deactivated.

Reply:
What would you propose?